cmd/dp-dataset-exporter/main.go

```diff
@@ -63,9 +66,14 @@ func main() {
 	filterStore := filter.NewStore(config.FilterAPIURL, config.FilterAPIAuthToken, &httpClient)
 	observationStore := observation.NewStore(neo4jConnPool)
 	fileStore := file.NewStore(config.AWSRegion, config.S3BucketName)
-	eventProducer := event.NewAvroProducer(kafkaProducer)
+	eventProducer := event.NewAvroProducer(kafkaProducer, schema.CSVExportedEvent)
 
-	eventHandler := event.NewExportHandler(filterStore, observationStore, fileStore, eventProducer)
+	datasetAPICli := dataset.New(config.DatasetAPIURL)
+	datasetAPICli.SetInternalToken(config.DatasetAPIAuthToken)
+
+	generateFileID := uuid.NewV4().String()
```

Comment:
It seems that generating the file ID here will give the same ID to all generated downloads.

Reply:
It looks like the intention was to pass in an ID generating function. I have refactored to just generate a new UUID when generating the file. We can rework this later if we change the file ID's to something other than UUID's.